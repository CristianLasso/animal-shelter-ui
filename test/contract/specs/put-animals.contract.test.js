import {provider} from '../config/init-pact';  
import {Matchers} from '@pact-foundation/pact';
import {AnimalController} from '../../../controllers';

describe('Animal Service', () => {     
    describe('When a request to edit one animal is made', () => {         
        beforeAll(async () => {             
            await provider.setup();             
            await provider.addInteraction({     
                uponReceiving: 'a request to edit one animal',     
                state: "has the animal",     
                withRequest: {         
                    method: 'PUT',         
                    path: '/animals/manchas'     
                },     
                willRespondWith: {         
                    status: 200,         
                    body: Matchers.eachLike({         
                        id: Matchers.like(69),
                        name: Matchers.like('manchas'),                 
                        breed: Matchers.like("Toyger"),                 
                        gender: Matchers.like("Male"),                 
                        vaccinated: Matchers.boolean(true)             
                    })     
                } 
                });         
        });          
        test('should return the correct data', async () => {             
            const response = await AnimalController.updateAnimal("manchas");
            expect(response.data).toMatchSnapshot();  
            await provider.verify()       
        });          
        afterAll(() => provider.finalize());     
    }); 
});