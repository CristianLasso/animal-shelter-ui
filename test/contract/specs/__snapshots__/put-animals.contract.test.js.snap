// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animal Service When a request to edit one animal is made should return the correct data 1`] = `
Array [
  Object {
    "breed": "Toyger",
    "gender": "Male",
    "id": 69,
    "name": "manchas",
    "vaccinated": true,
  },
]
`;
